include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -27,7 +27,17 @@ template<typename T>
 T atomicAdd(NBL_REF_ARG(T) ptr, T value)
 {
     // FUTURE: assert the scalart type is an integer type, then assert device traits maybe? but default to traits there?
```

Comment:
the way to address it is to use `enable_if` checking that `is_same_v<T,uint32_t> || is_same<T,int32_t>` (add `_v` template variable to `type_traits` while you're at it

Reply:
also I'd probably have that `enable_if` on the spirv intrinsic itself if possible.

Reply:
Done